### DISCUSSION TITLE
Fail to load GROMACS trajectory

### DISCUSSION THREAD

**Discussion by horlust** [MDAnalysis/mdanalysis#4537] [Q&A] [answered]
Hello,

I am trying to load a GROMACS trajectory on MDAnalysis but I keep encountering this error:

```
# define universe with MDAnalysis
u  = mda.Universe('nvt_3.tpr', 'nvt_3.trr')
print(u)
print(len(u.trajectory))
```

---------------------------------------------------------------------------
NotImplementedError                       Traceback (most recent call last)
File ~/anaconda3/lib/python3.11/site-packages/MDAnalysis/core/universe.py:110, in _topology_from_file_like(topology_file, topology_format, **kwargs)
    109     with parser(topology_file) as p:
--> 110         topology = p.parse(**kwargs)
    111 except (IOError, OSError) as err:
    112     # There are 2 kinds of errors that might be raised here:
    113     # one because the file isn't present
    114     # or the permissions are bad, second when the parser fails

File ~/anaconda3/lib/python3.11/site-packages/MDAnalysis/topology/TPRParser.py:212, in TPRParser.parse(self, tpr_resid_from_one, **kwargs)
    211 try:
--> 212     th = tpr_utils.read_tpxheader(data)                    # tpxheader
    213 except (EOFError, ValueError):

File ~/anaconda3/lib/python3.11/site-packages/MDAnalysis/topology/tpr/utils.py:233, in read_tpxheader(data)
    232 fileVersion = data.unpack_int()  # version of tpx file
--> 233 fileVersion_err(fileVersion)
    235 # This is for backward compatibility with development version 77-79 where
    236 # the tag was, mistakenly, placed before the generation.
    237 # These versions are development versions between the 4.5 and 4.6 series.

File ~/anaconda3/lib/python3.11/site-packages/MDAnalysis/topology/tpr/utils.py:208, in fileVersion_err(fver)
    207 if fver not in setting.SUPPORTED_VERSIONS:
--> 208     raise NotImplementedError(
    209         f"Your tpx version is {fver}, which this parser does not support, yet "
    210     )

NotImplementedError: Your tpx version is 133, which this parser does not support, yet 

During handling of the above exception, another exception occurred:

ValueError                                Traceback (most recent call last)
Cell In[9], line 2
      1 # define universe with MDAnalysis
----> 2 u  = mda.Universe('nvt_3.tpr', 'nvt_3.trr')
      3 print(u)
      4 print(len(u.trajectory))

File ~/anaconda3/lib/python3.11/site-packages/MDAnalysis/core/universe.py:356, in Universe.__init__(self, topology, all_coordinates, format, topology_format, transformations, guess_bonds, vdwradii, fudge_factor, lower_bound, in_memory, in_memory_step, *coordinates, **kwargs)
    354 if not isinstance(topology, Topology) and not topology is None:
    355     self.filename = _check_file_like(topology)
--> 356     topology = _topology_from_file_like(self.filename,
    357                                         topology_format=topology_format,
    358                                         **kwargs)
    360 if topology is not None:
    361     self._topology = topology

File ~/anaconda3/lib/python3.11/site-packages/MDAnalysis/core/universe.py:125, in _topology_from_file_like(topology_file, topology_format, **kwargs)
    121         raise IOError("Failed to load from the topology file {0}"
    122                         " with parser {1}.\n"
    123                         "Error: {2}".format(topology_file, parser, err))
    124 except (ValueError, NotImplementedError) as err:
--> 125     raise ValueError(
    126         "Failed to construct topology from file {0}"
    127         " with parser {1}.\n"
    128         "Error: {2}".format(topology_file, parser, err))
    129 return topology

ValueError: Failed to construct topology from file nvt_3.tpr with parser <class 'MDAnalysis.topology.TPRParser.TPRParser'>.
Error: Your tpx version is 133, which this parser does not support, yet 

It is the first time I encounter this error. Do you have any idea what it might be?

Thanks!

**Comment by hmacdope** [comment, marked as answer]
@horlust this should have been fixed by #4523, can you try a fresh install of `develop`?

**Reply by horlust** [reply]
Sorry, I read the discussion but didn't understand what to do exactly...

**Reply by hmacdope** [reply]
You should just be able to clone the repo and run a fresh install of MDAnalysis with `pip install package`  and it should just work.

**Reply by horlust** [reply]
I did git clone https://github.com/MDAnalysis/mdanalysis.git and then pip install mdanalysis inside the folder and it still doesn't work. I also tried git clone -b develop https://github.com/MDAnalysis/mdanalysis.git. I think the problem must be me but I don't know what I'm doing wrong...

**Reply by horlust** [reply]
I think I managed to install it. I did pip install -e package. However, now I cannot define the universe, I get:

---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
Cell In[13], line 2
      1 # define universe with MDAnalysis
----> 2 u  = mda.Universe('nvt_3.tpr', 'nvt_3.trr')
      3 print(u)
      4 print(len(u.trajectory))

AttributeError: module 'MDAnalysis' has no attribute 'Universe'

Still trying to find what's wrong...

**Reply by orbeckst** [reply]
This sounds as if you messed up your python package environment or you're running inside the mdanalysis repository. 
1. make sure that you're NOT launching jupyter, ipython, or python anywhere under `mdanalysis`
2. I recommend to follow the developer instructions for setting up a development environment https://userguide.mdanalysis.org/stable/contributing_code.html ; ultimately it's extremely useful to learn how to use separate environments (either with conda or virtualenv) and create a specific development environment version for a development version of MDAnalysis. I'd set up a completely fresh environment for develop and then install into this env with `pip install package/`.

**Reply by horlust** [reply]
Hi Oliver,

I have followed the steps to create a conda environment. Then, in the mdanalysis environment and folder, I did pip install -e package and it worked succesfully. However, I still have the same problem. I am not running the code inside the mdanalysis repository. It seems like the code doesn't find most of the attributes, like __version__ or Universe - when I do mda.() and press tab I don't find "Universe". I'll keep trying :(

**Reply by orbeckst** [reply]
What operating system are you using? Which version of Python?

If you create a fresh environment and show *every single command* then we might be able to point out issues. 

When you activate your environment and open `Python` and then execute
```python 
import numpy as np 
print(np.__version__)
print(np.__file__)

import MDAnalysis as mda 
print(mda.__version__)
print(mda.__file__)
```
then what do you get? (I think __file__ should work… I’m typing on a phone and can’t try it out right now).

**Reply by horlust** [reply]
I am using Linux, Python version is 3.11.

I first update conda with `conda update conda`. Then I do `conda create --name mdanalysis-dev "python>=3.9"` and activate the env with conda `conda activate mdanalysis-dev`. Then I go to the cloned repository with `cd mdanalysis`. I do `pip install package/`. Installation is succesful. Inside the env, I go to the folder in which I save my codes and open jupyter notebook. The first cell of one of my codes is:

```
import MDAnalysis as mda

import warnings
# suppress some MDAnalysis warnings about PSF files
warnings.filterwarnings('ignore')

%matplotlib inline
import numpy as np
```

But I get:

---------------------------------------------------------------------------
ModuleNotFoundError                       Traceback (most recent call last)
/tmp/ipykernel_12901/3397896404.py in <module>
----> 1 import MDAnalysis as mda
      2 
      3 import warnings
      4 # suppress some MDAnalysis warnings about PSF files
      5 warnings.filterwarnings('ignore')

ModuleNotFoundError: No module named 'MDAnalysis'

If I run your code, I get:

```
1.21.5
/usr/lib/python3/dist-packages/numpy/__init__.py
```
---------------------------------------------------------------------------
ModuleNotFoundError                       Traceback (most recent call last)
/tmp/ipykernel_12901/3741928057.py in <module>
      3 print(np.__file__)
      4 
----> 5 import MDAnalysis as mda
      6 print(mda.__version__)
      7 print(mda.__file__)

ModuleNotFoundError: No module named 'MDAnalysis'

One more thing, the python version of the conda env is 3.12.2.

Thanks for your help! I still don't know what I'm doing wrong...

**Reply by orbeckst** [reply]
Try my commands in the python interpreter and not in Jupyter. 

Also, what do the commands 
```
which python
which jupyter
```
show?

**Reply by orbeckst** [reply]
Also, did you install the dependencies in to your environment following https://userguide.mdanalysis.org/stable/contributing_code.html#id1  ?

**Reply by horlust** [reply]
The output from the python interpreter is:

```
>>> import numpy as np 
>>> print(np.__version__)
1.26.4
>>> print(np.__file__)
/home/maria/anaconda3/envs/mdanalysis-dev/lib/python3.12/site-packages/numpy/__init__.py
>>> import MDAnalysis as mda 
>>> print(mda.__version__)
2.8.0-dev0
>>> print(mda.__file__)
/home/maria/anaconda3/envs/mdanalysis-dev/lib/python3.12/site-packages/MDAnalysis/__init__.py
```

The output for which python and which jupyter is:

```
(base) maria@PCGEC243:~$ conda activate mdanalysis-dev
(mdanalysis-dev) maria@PCGEC243:~$ which python
/home/maria/anaconda3/envs/mdanalysis-dev/bin/python
(mdanalysis-dev) maria@PCGEC243:~$ which jupyter
/usr/bin/jupyter
```

**Reply by horlust** [reply]
Yep, I did:

```
conda install -c conda-forge \
  'Cython>=0.28' \
  'numpy>=1.21.0' \
  'biopython>=1.80' \
  'networkx>=2.0' \
  'GridDataFormats>=0.4.0' \
  'mmtf-python>=1.0.0' \
  'joblib>=0.12' \
  'scipy>=1.5.0' \
  'matplotlib>=1.5.1' \
  'tqdm>=4.43.0' \
  'threadpoolctl'\
  'packaging' \
  'fasteners' \
  'netCDF4>=1.0' \
  'h5py>=2.10' \
  'chemfiles>=0.10' \
  'pyedr>=0.7.0' \
  'pytng>=0.2.3' \
  'gsd>3.0.0' \
  'rdkit>=2020.03.1' \
  'parmed' \
  'seaborn' \
  'scikit-learn' \
  'tidynamics>=1.0.0' \
  'mda-xdrlib'

# documentation dependencies
conda install -c conda-forge 'mdanalysis-sphinx-theme>=1.3.0' docutils sphinx-sitemap sphinxcontrib-bibtex pybtex pybtex-docutils
python -m pip install docutils sphinx-sitemap sphinxcontrib-bibtex pybtex pybtex-docutils

```
It seems like it doesn't work on jupyter but it does on the python interpreter.

**Reply by orbeckst** [reply]
As you can see, inside `Python` you can find MDAnalysis! 

The problem is that you are using Jupyter *outside your environment*. You *must also install Jupyter into the same environment where all your other packages are*. Otherwise Jupyter (or ipython) won’t be able to find packages inside the MDA-dev environment. 

This is a very common issue when people use Jupyter and environments. The basic rule is to always install everything you need into your environment and to never try to use anything outside. You can use the commands shown above to see where exactly python modules or executables are located: they should *all have a path under your envronment*.

**Reply by horlust** [reply]
Oh silly me. Thanks, I think it worked! Thank you very much for your time :)

**Reply by orbeckst** [reply]
Glad that it worked! Could you then mark @hmacdope ‘s post as the answer to your question—-assuming that the MDA dev version now can read your new gmx files?

**Reply by horlust** [reply]
Done, thanks :)
